lib/moonshot/shell.rb

```diff
@@ -51,4 +61,23 @@ def sh_step(cmd, args = {})
       step.success
     end
   end
+
+  # Retries every second upto maximum of 5 minutes
+  # with the default options.
+  #
+  # @param cmd [String] command to execute.
+  # @param fail [Boolean] Raise error on failure.
+  # @param stdin [String] Input to the command.
+  # @param opts [Hash] Options for retriable.
+  #
+  # @return [String] Stdout form the command.
+  def sh_retry(cmd, fail: true, stdin: '', opts: {})
+    Retriable.retriable(DEFAULT_RETRY_OPTIONS.merge(opts)) do
+      out = sh_out(cmd, stdin: stdin)
+      yield out if block_given?
+      out
+    end
+  rescue RuntimeError => e
```

Comment:
This is an interesting situation right here, cause even though `sh_out` can only raise `RuntimeError`, the function caller can raise _any_ exception from the block it has passed to the function.